### PR TITLE
IEP-787: Replace the CMake change with a project description reader to find project binaries.

### PR DESCRIPTION
**Pull request by sigmaaa**
## Description

For now, we are changing the CMake file after project creation, so project binaries will always have the same name as the project, but with this approach, some templates don't work. 

This PR contains a refactoring of the places where we look for binaries and a replacement based on calling the project name with the value from project_decsription.json. 
As example:
```
elfFilePath = project.getProject().getFolder("build").getFile(project.getName().replace(" ", "_").concat(".elf")).getLocation() 
.toString();

```
replaced with:

```
IFile elfFile = new ProjectDescriptionReader(project.getProject()).getAppElfFile();
elfFilePath = elfFile == null ? null : elfFile.getLocation().toString();
```

Fixes # ([IEP-787](https://jira.espressif.com:8443/browse/IEP-787))

## Type of change

Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)

## How has this been tested?

So far I have found only two places where these changes have had an impact. Application Level Tracing and Heap Tracing, so tests were next:

- Test application level tracing feature with different project names
- Test heap tracing feature with different project names

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Application Level Tracing
- Heap Tracing
- some other places, where we are looking for binaries?

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

### CODE REVIEW

File: bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/ProjectDescriptionReader.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	private JSONObject read()
+	{
+		IPath pathToProjectDescriptionJsonFilePath = Optional.ofNullable(project)
+				.map(p -> p.getFolder(IDFConstants.BUILD_FOLDER)).map(b -> b.getFile("project_description.json")) //$NON-NLS-1$
+				.map(IResource::getLocation).orElse(new Path(StringUtil.EMPTY));
+		File projectDescriptionJsonFile = pathToProjectDescriptionJsonFilePath.toFile();
+		JSONParser parser = new JSONParser();
+		try (FileReader reader = new FileReader(projectDescriptionJsonFile))
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/ProjectDescriptionReader.java:53]: Can we use GenericJsonReader to read this as we already using this here https://github.com/espressif/idf-eclipse-plugin/blob/f0304d96aa3692b784b088aefbd22fd4bb4effcd/bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/IDFUtil.java#L531

**Review by kolipakakondal — CHANGES_REQUESTED** [review]
LGTM.

**Comment by kolipakakondal** [comment]
Hi @AndriiFilippov could you please check this

**Comment by AndriiFilippov** [comment]
Hi @sigmaaa ! 

Tested under: 
OS: Windows 10
ESP-IDF: v5.0/master 

LGTM. 👍

**Pull request merged**